ddl/table_test.go

```diff
@@ -417,6 +418,6 @@ func (s *testTableSuite) TestTableResume(c *C) {
 		Type:       model.ActionDropTable,
 		BinlogInfo: &model.HistoryInfo{},
 	}
-	testRunInterruptedJob(c, d, job)
+	c.Assert(d.doDDLJob(mock.NewContext(), job), IsNil)
```

Comment:
This test effect is different from the original function of `testRunInterruptedJob`. I think we can do https://github.com/pingcap/tidb/pull/18844/files#diff-4226364954d30f0f8c2648cb19da228eL235 like https://github.com/pingcap/tidb/pull/18844/files#diff-1461e7e6b6b324ad06052b4709e8c1daR87.

Reply:
now `testRunInterruptedJob` keeps same to caller, and use `*ddl.mu` to avoid data race